In [1]:
import pandas as pd

In [2]:
sales = pd.read_csv("sales.csv")

In [3]:
%%sql
SELECT
    product_line,
    SUM(quantity) AS total_quantity
FROM
    sales
GROUP BY
    product_line
ORDER BY
    total_quantity DESC;

product_line  total_quantity
0  Suspension & traction          2145.0
1        Breaking system          2130.0
2      Electrical system          1698.0
3           Frame & body          1619.0
4          Miscellaneous          1176.0
5                 Engine           627.0

In [4]:
%%sql
SELECT
    product_line,
    SUM(quantity) AS total_quantity,
    COUNT(DISTINCT strftime('%Y-%m', CAST(date AS DATE))) AS months_active
FROM
    sales
GROUP BY
    product_line
ORDER BY
    total_quantity DESC;

product_line  total_quantity  months_active
0  Suspension & traction          2145.0              3
1        Breaking system          2130.0              3
2      Electrical system          1698.0              3
3           Frame & body          1619.0              3
4          Miscellaneous          1176.0              3
5                 Engine           627.0              3

In [5]:
%%sql
SELECT
    product_line,
    SUM(quantity) AS total_quantity
FROM
    sales
GROUP BY
    product_line
ORDER BY
    total_quantity ASC;

product_line  total_quantity
0                 Engine           627.0
1          Miscellaneous          1176.0
2           Frame & body          1619.0
3      Electrical system          1698.0
4        Breaking system          2130.0
5  Suspension & traction          2145.0

# Product Line Analysis

This report provides an analysis of the top-selling product lines based on the provided dataset.

## Overview

The Product Line Analysis aims to identify the top-selling product lines and assess their performance during the specified period.

## Findings

### Top-Selling Product Lines

Based on the dataset, the top-selling product lines during the specified period are as follows:

1. **Suspension & Traction**: Total quantity sold: 2145 units.
2. **Breaking System**: Total quantity sold: 2130 units.
3. **Electrical System**: Total quantity sold: 1698 units.
4. **Frame & Body**: Total quantity sold: 1619 units.
5. **Miscellaneous**: Total quantity sold: 1176 units.
6. **Engine**: Total quantity sold: 627 units.

### Observations

- The top-selling product lines include various components related to vehicle systems and maintenance, such as suspension & traction, breaking system, electrical system, and frame & body.
- Miscellaneous products also contribute significantly to total sales, indicating a diverse range of products sold during the specified period.